test/Main.hs

```diff
@@ -107,11 +116,35 @@ resizeIdempotence =
 
     resize = fromStreamD . resizeD . toStreamD
 
+addUncompressedHeader :: Array.Array Word8 -> IO (Array.Array Word8)
+addUncompressedHeader arr = do
+    let negLen = -1 * unsafeIntToI32 (Array.byteLength arr)
+        hdr = Array.fromList [negLen, negLen]
```

Comment:
Uncompressed length should be positive. How is this working? It seems we are not doing any checks on that anywhere, in this code path? We should catch this somewhere.